convert single task to ES6

CKrawczyk opened the pull request:
Converts the `single` task to ES6

# Review Checklist

- [x] Does it work in all major browsers: Firefox, Chrome, Edge, Safari?
- [x] Does it work on mobile?
- [x] Can you `rm -rf node_modules/ && npm install` and app works as expected?
- [x] Did you deploy a staging branch? https://convert-single-task.pfe-preview.zooniverse.org/
- [ ] Did you get any type checking errors from [Babel Typecheck](https://github.com/codemix/babel-plugin-typecheck)


## Optional

- [x] If it's a new component, is it in ES6? Is it clear of warnings from ESLint?
- [ ] Have you replaced any `ChangeListener` or `PromiseRenderer` components with code that updates component state?
- [x] If changes are made to the classifier, does the dev classifier still work?
- [ ] Have you added in [flow type annotations](https://flowtype.org/docs/type-annotations.html)?

eatyourgreens commented on app/classifier/tasks/single.jsx:
```diff
@@ -0,0 +1,165 @@
+
+Summary.defaultProps = {
+  task: null,
+  annotation: null,
```
This default value will throw an error when checking properties eg. `if annotation.value`. A better default might be an empty annotation object, if that doesn't break the code.

CKrawczyk replied:
Isn't that taken care of by the `getDefaultAnnotation` method?  Although I don't mind changing the default prop to `{ value: null }`.

eatyourgreens replied:
Oh yeah, I missed that. So tasks define a default value, then a method that completely ignores that value in favour of something else?

eatyourgreens replied:
Oh wait, I was referring to the default on the `Summary` component, rather than the task.

CKrawczyk replied:
Oh, good call.  As for the task components, the `getDefaultAnnotation` method is called via the `preparetoclassify` method on mount (https://github.com/zooniverse/Panoptes-Front-End/blob/master/app/classifier/index.cjsx#L76, https://github.com/zooniverse/Panoptes-Front-End/blob/master/app/classifier/index.cjsx#L143, https://github.com/zooniverse/Panoptes-Front-End/blob/master/app/classifier/index.cjsx#L499), so the `defaultProp` will never be used.  By extension the `Summary` component should always have a non-null annotation object.

It might be in order to remove the `Summary`'s default annotation and just make the prop `isRequired`.

eatyourgreens replied:
Good idea. I like that, if it doesn't break the code ie. we can rely on the classifier always passing annotations.

eatyourgreens replied:
By the way, you can also use [`React.PropTypes.shape`](https://facebook.github.io/react/docs/typechecking-with-proptypes.html) to enforce that the annotation should have a value.

CKrawczyk commented:
The staging link has been updated.

CKrawczyk commented:
I found and fixed a bug introduced in the `isAnnotationComplete` function (returned false when value was `0`) and added tests catch this bug.  The staged link has been updated.
